test/compilable/test20810.d

```diff
@@ -0,0 +1,126 @@
+/*
+REQUIRED_ARGS: --help
```

Comment:
don't use `--help` but pass a simple module containing an empty main func. The use of `--help` is, in addition of not testing the right thing, something that would have to be maintained over years.

Reply:
> don't use `--help` but pass a simple module containing an empty main func. The use of `--help` is, in addition of not testing the right thing.

Not an ideal test but running `dmd --help` triggers that bug.

> something that would have to be maintained over years.

Only the current version, I'll re-enable a maintenance free version after this PR is merged.

Reply:
A brittle integration test for this is overkill; a trivial little unittest would do the job perfectly, just like https://github.com/dlang/dmd/blob/77ae1622340762d776fb5548b8cfbe27a6abd593/src/dmd/root/filename.d#L363-L370

Reply:
Agreed, a unittest is the best solution here.